core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -315,10 +326,10 @@ public void onLoad(Run<?, ?> r) {
         for (ParameterValue v : this.parameters) {
             if (this.parameterDefinitionNames.contains(v.getName()) || isSafeParameter(v.getName())) {
                 filteredParameters.add(v);
-            } else {
+            } else if ("false".equalsIgnoreCase(shouldKeepFlag)) {
                 LOGGER.log(Level.WARNING, "Skipped parameter `{0}` as it is undefined on `{1}`. Set `-D{2}=true` to allow "
                         + "undefined parameters to be injected as environment variables or `-D{3}=[comma-separated list]` to whitelist specific parameter names, "
-                        + "even though it represents a security breach",
+                        + "even though it represents a security breach or `-D{2}=false to explicitly forbid it without warning.",
```

Comment:
Misses closing backtick.

Would make it a second sentence: Set `-D{2}=false` to no longer show this message.